gzip.go

```diff
@@ -63,31 +105,29 @@ func acceptsGzip(r *http.Request) bool {
 // works.
 //
 // See: http://tools.ietf.org/html/rfc2616#section-14.3
-func parseEncodings(s string) (codings, error) {
+func parseEncodings(s string) (codings, ErrorList) {
```

Comment:
I think it'd be more helpful to the average caller for the return type here to be the `error` interface. That way, they can pass the error along as-is if they're not interested in what happened, but can still cast it back to an ErrorList if they want to do something more sophisticated with it. ErrorList doesn't currently implement `error`, but it should be simple enough to make it do so. The [Error handing and Go](http://blog.golang.org/error-handling-and-go) blog post contains some good examples of this kind of thing.